### PR TITLE
DEVENGAGE-1557:  Updated to latest Go SDK. . Added the media_region, …

### PR DESCRIPTION
**Pull request by carnellj-genesys**
…caller_id and caller_name attributes

### CODE REVIEW

**Review by dginty4 — COMMENTED** [review]
More of the matrix tests seem to be failing for this than usual

File: genesyscloud/resource_genesyscloud_telephony_providers_edges_site.go [hunk 1]
```diff
@@ -282,6 +328,11 @@ func createSite(ctx context.Context, d *schema.ResourceData, meta interface{}) d
 		return diag.Errorf("Location with id %v does not have an emergency number", locationId)
 	}
 
+	err = validateMediaRegions(mediaRegions, sdkConfig)
```
Comment (carnellj-genesys) [inline genesyscloud/resource_genesyscloud_telephony_providers_edges_site.go:331]: This is the new validator function I added to the code.  I could not make a validator function in the schema yet as the Terraform framework does not support validation on lists and sets.

File: genesyscloud/validators.go [hunk 2]
```diff
@@ -12,10 +12,15 @@ import (
 
 func validatePhoneNumber(number interface{}, _ cty.Path) diag.Diagnostics {
 	if numberStr, ok := number.(string); ok {
-		_, err := phonenumbers.Parse(numberStr, "US")
+		phoneNumber, err := phonenumbers.Parse(numberStr, "US")
 		if err != nil {
 			return diag.Errorf("Failed to validate phone number %s: %s", numberStr, err)
 		}
+
+		formattedNum := phonenumbers.Format(phoneNumber, phonenumbers.E164)
```
[the remark this answers is not in the document]
Comment (dginty4) [reply genesyscloud/validators.go:21]: As you said, a lot of the tests are now failing due to invalid phone numbers. There was also some 'plan not empty' errors related to the media_regions

File: genesyscloud/resource_genesyscloud_telephony_providers_edges_site.go [hunk 3]
```diff
@@ -494,6 +584,22 @@ func deleteSite(ctx context.Context, d *schema.ResourceData, meta interface{}) d
 	})
 }
 
+func buildMediaRegionList(d *schema.ResourceData) *[]string {
```
Comment (charliecon) [inline genesyscloud/resource_genesyscloud_telephony_providers_edges_site.go:587]: The function `buildSdkStringListFromInterfaceArray` does this job I'm fairly sure.

**Comment by charliecon** [comment]
The binary was accidentally checked in

File: genesyscloud/resource_genesyscloud_telephony_providers_edges_site.go [hunk 4]
```diff
@@ -262,6 +280,31 @@ func siteExporter() *ResourceExporter {
...
+
+	homeRegion := telephonyRegions.AwsHomeRegion
+	coreRegions := telephonyRegions.AwsCoreRegions
+	satRegions := telephonyRegions.AwsSatelliteRegions
+
+	for _, region := range *regions {
+		if region != *homeRegion &&
+			!StringInSlice(region, *coreRegions) &&
+			!StringInSlice(region, *satRegions) {
+			fmt.Errorf("Region %s is not a valid media region.  Please refer to the Genesys Cloud GET /api/v2/telephony/mediaregions for list of valid regions.", regions)
```
[the remark this answers is not in the document]
Comment (carnellj-genesys) [reply genesyscloud/resource_genesyscloud_telephony_providers_edges_site.go:300]: Excellent catch.  Fixed

[the remark this answers is not in the document]
Comment (carnellj-genesys) [reply genesyscloud/resource_genesyscloud_telephony_providers_edges_site.go]: Done

**Review by charliecon — APPROVED** [review]

**Pull request merged**